src/ecs/mod.rs

```diff
@@ -7,14 +7,15 @@ pub mod transform;
 pub mod rendering;
 pub mod audio;
 
+use app::ApplicationBuilder;
 use error::Result;
 
-/// Extension trait for all systems.
-pub trait SystemExt<'a, A>: System<'a> {
-    /// Constructs a new system with the given arguments.
-    /// Registers all supported components with the World.
-    /// Puts resources into World.
-    fn build(args: A, world: &mut World) -> Result<Self>
-    where
-        Self: Sized;
+/// Describes a bundle of ECS components, resources and systems
+pub trait ECSBundle<'a, 'b, T: ::state::State + 'a> {
```

Comment:
This bound shouldn't be here. Please, remove it from `ApplicationBuilder`'s functions where `T: State` is not required. I believe the only function that requires `T: State` is `ApplicationBuilder::build`

Reply:
Well, ApplicationBuilder requires it at the moment, and that's a separate change from this PR imho.

Reply:
I fixed it.